torch/distributed/_composable/checkpoint_activation.py

```diff
@@ -61,16 +63,45 @@ def checkpoint(module: nn.Module) -> nn.Module:
     """
     torch._C._log_api_usage_once("torch.distributed.checkpoint")
 
-    def forward_pre_hook(module: nn.Module, inputs: Tuple[Any, ...]) -> None:
+    kwargs = kwargs or {}
```

Comment:
nit: I think we do not need this; if you have `**kwargs` in the signature, then Python will make sure `kwargs` is `{}` if nothing is passed. I think for things like `__torch_dispatch__` we need this because `kwargs` defaults to `None`.

Reply:
Ohh thanks! I didn't know this. Btw why does it default to `None` for `__torch_dispatch__`?

Reply:
> Btw why does it default to None for __torch_dispatch__?

I am not too sure. It might be somehow related to the arg parsing logic.